OsmAnd/src/net/osmand/plus/views/layers/GPXLayer.java

```diff
@@ -999,6 +1018,70 @@ private SelectedGpxPoint createSelectedGpxPoint(SelectedGpxFile selectedGpxFile,
 		return new SelectedGpxPoint(selectedGpxFile, projectionPoint, prevPoint, nextPoint, bearing);
 	}
 
+	private GPXFile createGpxFileFromRenderedObject(NativeLibrary.RenderedObject object) {
+		OsmandApplication app = view.getApplication();
+		GPXFile gpxFile = new GPXFile(Version.getFullVersion(app));
+		String fileName = object.getFileNameByExtension(IndexConstants.GPX_FILE_EXT);
+		gpxFile.path = app.getAppPath(IndexConstants.GPX_FROM_OBF_DIR).getPath() + "/" + fileName;
+		GPXUtilities.Track track = new GPXUtilities.Track();
+		String refValue = object.getTagValue(TravelObfHelper.REF_TAG);
+
+		List<BinaryMapDataObject> segmentsObjects = new ArrayList<>();
```

Comment:
same code was part of travelobfhelper, so it should be there probably and combined with existing code